### ISSUE TITLE
How to enable HTTPS for Presto JDBC

### ISSUE THREAD

**Issue by nico-arianto** [serge-rider/dbeaver#2228]
Case: Connect to Presto with LDAP authentication which required for DBeaver + JDBC to connect the API with HTTPS.

Issue: There's no option to enable SSL.

Tested with __maven:/com.facebook.presto:presto-jdbc:0.170__

Info: We did try to set `Connection Properties` -> `SSL: True`, but doesn't work because the Dbeaver still accessing the API with HTTP.

Reference:
* [Presto JDBC Driver](https://prestodb.io/docs/current/installation/jdbc.html)

**Comment by serge-rider** [comment]
1. I think I'll add Presto driver config in default DBeaver configuration.
2. I used latest driver (0.185) for testing
3. Setting (manually adding user property) property `SSL` to `true` in connection driver properties seems to be working

![image](https://user-images.githubusercontent.com/6398845/31044806-45e5eb3e-a5df-11e7-8cbc-8e34808a43fe.png)

**Comment by nico-arianto** [comment]
Don't get misleading with this weird false Success, because that status (refer to `Server: Presto ?`) means failed.

If it able to connect successfully, it will show you the Presto Info without the `?` character.

You can try to open the connection and expand one of the tables it will show you the real error which mostly it cannot access the Presto API as it's still using HTTP (not HTTPS).

Info: I also not sure if this false Success notification is wrongly with the JDBC driver or DBeaver it self.

**Comment by serge-rider** [comment]
It is failed in my case because I don't have Presto server locally. I've just check that adding ssl parameter makes driver happy (it doesn't complain on security issue).

But how do you know that it uses HTTP? Do you see any specific errors? Maybe in debug/error log? (https://github.com/serge-rider/dbeaver/wiki/Log-files)

**Comment by serge-rider** [comment]
In regards to success connection issue. This is PrestoDB weird behavior. It creates Connection object in any case (by JDBC API this means connection was successful). 
But any following queries are failing.

**Comment by nico-arianto** [comment]
I tested again in my local with the latest driver as you suggested (0.185) and it works nicely. Thanks!

Looks like there is an issue with Presto JDBC 0.170, it's ignoring the `SSL` properties.

We did use Presto JDBC 0.170 because we are using AWS EMR Presto 0.170 and our initial intention is to avoid any incompatibility issue between the server and client.

This is the error that we got when using Presto JDBC 0.170 + SSL properties set:
```
!ENTRY org.jkiss.dbeaver.model 4 0 2017-10-01 10:42:12.711
!MESSAGE Error executing query
!STACK 0
java.sql.SQLException: Error executing query
	at com.facebook.presto.jdbc.PrestoStatement.execute(PrestoStatement.java:232)
	at com.facebook.presto.jdbc.PrestoStatement.executeQuery(PrestoStatement.java:69)
	at com.facebook.presto.jdbc.PrestoDatabaseMetaData.select(PrestoDatabaseMetaData.java:1416)
	at com.facebook.presto.jdbc.PrestoDatabaseMetaData.getTables(PrestoDatabaseMetaData.java:929)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCDatabaseMetaDataImpl.getTables(JDBCDatabaseMetaDataImpl.java:1119)
	at org.jkiss.dbeaver.ext.generic.model.TableCache.prepareLookupStatement(TableCache.java:85)
	at org.jkiss.dbeaver.ext.generic.model.TableCache.prepareLookupStatement(TableCache.java:1)
	at org.jkiss.dbeaver.model.impl.jdbc.cache.JDBCStructLookupCache.prepareObjectsStatement(JDBCStructLookupCache.java:132)
	at org.jkiss.dbeaver.model.impl.jdbc.cache.JDBCObjectCache.loadObjects(JDBCObjectCache.java:99)
	at org.jkiss.dbeaver.model.impl.jdbc.cache.JDBCObjectCache.getAllObjects(JDBCObjectCache.java:69)
	at org.jkiss.dbeaver.ext.generic.model.GenericObjectContainer.getTables(GenericObjectContainer.java:134)
	at org.jkiss.dbeaver.ext.generic.model.GenericObjectContainer.getPhysicalTables(GenericObjectContainer.java:117)
	at org.jkiss.dbeaver.ext.generic.model.GenericDataSource.getPhysicalTables(GenericDataSource.java:324)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.extractPropertyValue(DBNDatabaseNode.java:802)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.loadTreeItems(DBNDatabaseNode.java:488)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.loadChildren(DBNDatabaseNode.java:412)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:199)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:1)
	at org.jkiss.dbeaver.ui.navigator.NavigatorUtils.getNodeChildrenFiltered(NavigatorUtils.java:560)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:49)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:1)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:86)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:71)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:95)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
Caused by: com.facebook.presto.jdbc.internal.jetty.client.HttpResponseException: HTTP protocol violation: bad response on HttpConnectionOverHTTP@381c246f(l:/192.168.0.102:50102 <-> r:emr.techinasia.com/52.90.180.66:8449,closed=false)[HttpChannelOverHTTP@2edf102b(exchange=HttpExchange@35285d13 req=TERMINATED/null@null res=PENDING/null@null)[send=HttpSenderOverHTTP@820fdb3(req=QUEUED,snd=COMPLETED,failure=null)[HttpGenerator@51c68f84{s=START}],recv=HttpReceiverOverHTTP@70a9b386(rsp=IDLE,failure=null)[HttpParser{s=CLOSE,0 of 0}]]]
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpReceiverOverHTTP.badMessage(HttpReceiverOverHTTP.java:304)
	at com.facebook.presto.jdbc.internal.jetty.http.HttpParser.parseNext(HttpParser.java:1420)
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpReceiverOverHTTP.parse(HttpReceiverOverHTTP.java:159)
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:120)
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:70)
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:90)
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:115)
	at com.facebook.presto.jdbc.internal.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:273)
	at com.facebook.presto.jdbc.internal.jetty.io.FillInterest.fillable(FillInterest.java:95)
	at com.facebook.presto.jdbc.internal.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93)
	at com.facebook.presto.jdbc.internal.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303)
	at com.facebook.presto.jdbc.internal.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at com.facebook.presto.jdbc.internal.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
	at com.facebook.presto.jdbc.internal.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at com.facebook.presto.jdbc.internal.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:748)
!SUBENTRY 1 org.jkiss.dbeaver.model 4 0 2017-10-01 10:42:12.711
!MESSAGE HTTP protocol violation: bad response on HttpConnectionOverHTTP@381c246f(l:/192.168.0.102:50102 <-> r:emr.techinasia.com/52.90.180.66:8449,closed=false)[HttpChannelOverHTTP@2edf102b(exchange=HttpExchange@35285d13 req=TERMINATED/null@null res=PENDING/null@null)[send=HttpSenderOverHTTP@820fdb3(req=QUEUED,snd=COMPLETED,failure=null)[HttpGenerator@51c68f84{s=START}],recv=HttpReceiverOverHTTP@70a9b386(rsp=IDLE,failure=null)[HttpParser{s=CLOSE,0 of 0}]]]
!STACK 0
com.facebook.presto.jdbc.internal.jetty.client.HttpResponseException: HTTP protocol violation: bad response on HttpConnectionOverHTTP@381c246f(l:/192.168.0.102:50102 <-> r:emr.techinasia.com/52.90.180.66:8449,closed=false)[HttpChannelOverHTTP@2edf102b(exchange=HttpExchange@35285d13 req=TERMINATED/null@null res=PENDING/null@null)[send=HttpSenderOverHTTP@820fdb3(req=QUEUED,snd=COMPLETED,failure=null)[HttpGenerator@51c68f84{s=START}],recv=HttpReceiverOverHTTP@70a9b386(rsp=IDLE,failure=null)[HttpParser{s=CLOSE,0 of 0}]]]
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpReceiverOverHTTP.badMessage(HttpReceiverOverHTTP.java:304)
	at com.facebook.presto.jdbc.internal.jetty.http.HttpParser.parseNext(HttpParser.java:1420)
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpReceiverOverHTTP.parse(HttpReceiverOverHTTP.java:159)
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:120)
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:70)
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:90)
	at com.facebook.presto.jdbc.internal.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:115)
	at com.facebook.presto.jdbc.internal.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:273)
	at com.facebook.presto.jdbc.internal.jetty.io.FillInterest.fillable(FillInterest.java:95)
	at com.facebook.presto.jdbc.internal.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93)
	at com.facebook.presto.jdbc.internal.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303)
	at com.facebook.presto.jdbc.internal.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at com.facebook.presto.jdbc.internal.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
	at com.facebook.presto.jdbc.internal.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
	at com.facebook.presto.jdbc.internal.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
	at java.lang.Thread.run(Thread.java:748)
```
